components/org.wso2.carbon.identity.application.authenticator.samlsso/src/test/java/org/wso2/carbon/identity/application/authenticator/samlsso/util/SSOUtilsTest.java

```diff
@@ -0,0 +1,185 @@
+        Map<String, String> properties = new HashMap<>();
+        Assert.assertEquals(false, SSOUtils.isLogoutRequestSigned(properties));
+
+        properties.put(IdentityApplicationConstants.Authenticator.SAML2SSO.IS_LOGOUT_REQ_SIGNED, "false");
+        Assert.assertEquals(false, SSOUtils.isLogoutRequestSigned(properties));
+
+        properties.put(IdentityApplicationConstants.Authenticator.SAML2SSO.IS_LOGOUT_REQ_SIGNED, "true");
+        Assert.assertEquals(true, SSOUtils.isLogoutRequestSigned(properties));
+    }
+
+    @Test
+    public void testIsAuthnResponseSigned() {
+
+        Assert.assertEquals(false, SSOUtils.isAuthnResponseSigned(null));
+
+        Map<String, String> properties = new HashMap<>();
+        Assert.assertEquals(false, SSOUtils.isAuthnResponseSigned(properties));
+
+        properties.put(IdentityApplicationConstants.Authenticator.SAML2SSO.IS_AUTHN_RESP_SIGNED, "false");
+        Assert.assertEquals(false, SSOUtils.isAuthnResponseSigned(properties));
+
+        properties.put(IdentityApplicationConstants.Authenticator.SAML2SSO.IS_AUTHN_RESP_SIGNED, "true");
+        Assert.assertEquals(true, SSOUtils.isAuthnResponseSigned(properties));
+    }
+
+    @Test
+    public void testIsAssertionSigningEnabled() {
+
+        Assert.assertEquals(false, SSOUtils.isAssertionSigningEnabled(null));
+
+        Map<String, String> properties = new HashMap<>();
+        Assert.assertEquals(false, SSOUtils.isAssertionSigningEnabled(properties));
+
+        properties.put(IdentityApplicationConstants.Authenticator.SAML2SSO.IS_ENABLE_ASSERTION_SIGNING, "false");
+        Assert.assertEquals(false, SSOUtils.isAssertionSigningEnabled(properties));
+
+        properties.put(IdentityApplicationConstants.Authenticator.SAML2SSO.IS_ENABLE_ASSERTION_SIGNING, "true");
+        Assert.assertEquals(true, SSOUtils.isAssertionSigningEnabled(properties));
+    }
+
+    @Test
+    public void testIsAssertionEncryptionEnabled() {
+
+        Assert.assertEquals(false, SSOUtils.isAssertionEncryptionEnabled(null));
+
+        Map<String, String> properties = new HashMap<>();
+        Assert.assertEquals(false, SSOUtils.isAssertionEncryptionEnabled(properties));
+
+        properties.put(IdentityApplicationConstants.Authenticator.SAML2SSO.IS_ENABLE_ASSERTION_ENCRYPTION, "false");
+        Assert.assertEquals(false, SSOUtils.isAssertionEncryptionEnabled(properties));
+
+        properties.put(IdentityApplicationConstants.Authenticator.SAML2SSO.IS_ENABLE_ASSERTION_ENCRYPTION, "true");
+        Assert.assertEquals(true, SSOUtils.isAssertionEncryptionEnabled(properties));
+    }
+
+    @Test
+    public void testGetQueryMap() {
+
+        Assert.assertTrue(SSOUtils.getQueryMap("").isEmpty());
+
+        Map<String, String> queryParams = SSOUtils.getQueryMap(TestConstants.QUERY_STRING);
+
+        Assert.assertTrue(!queryParams.isEmpty());
+        Assert.assertEquals(queryParams.get("SAMLRequest"), "nVPRctowEPwVj97BDtCWaDAZCpMpM2mHAslD32T5jGVknauTCfn7yHZo" +
```

Comment:
We have these strings as constants defined ? Can we use them ? Or else let's define them ? Isn't that a good idea :)

Reply:
I have used a TestConstants class to define this kind of String constants. WDYT?